app/services/children_support_module/save_book_from_support_module_service.rb

```diff
@@ -0,0 +1,28 @@
+class ChildrenSupportModule
+  class SaveBookFromSupportModuleService
+		# Support module has books associated, but these can change over time
+		# So we save the book associated to a chosen module on the corresponding CSM before we send them
+
+		def initialize(group_id:)
+			@group_id = group_id
+    end
+
+    def call
+			# get not_programmed + with_support_module CSM for the group (only parent1 choices)
+			chosen_modules = ChildrenSupportModule.chosen_modules_for_group(@group_id)
+			chosen_modules = chosen_modules.uniq { |csm| [csm.child_id, csm.parent_id] }
+			# keep CSM of active children only
+			chosen_modules_ids = ChildrenSupportModule.joins(:child)
+													 .where(id: chosen_modules)
+													 .where(children: { group_status: 'active' }).ids.join(',')
```

Comment:
note to self : return s'il n'y a pas de chosen_modules_ids sinon la requête SQL risque d'échouer